mender-conversion-tool.sh

```diff
@@ -0,0 +1,688 @@
+           \nrootfs part size (sectors): ${prootfs_size}\
+           \ndata part size (sectors): ${pdata_size}.\n"
+}
+
+calculate_overall_size() {
+  unallocated=$(( $pboot_start * $sector_size ))
+  boot=$(( $pboot_size * $sector_size ))
+  rootfs=$(( 2 * ${prootfs_size} * $sector_size ))
+  data=$(( $pdata_size * $sector_size ))
+  overall=$(( $unallocated + $boot + $rootfs + $data ))
+
+  echo ${overall}
+}
+
+create_image() {
+  echo "2. Writing $overall bytes to .sdimg file..."
+  bs=$(( 1024*1024 ))
+  overall=$1
+  count=$(( ${overall} / ${bs} ))
+  dd if=/dev/zero of=${target_dev} bs=${bs} count=${count}
+}
+
+format_block_device() {
+  if [ -z "$1" ]; then
+    echo "Error: no root filesystem size supplied"
+    exit 1
+  fi
+
+  if [ -z "$2" ]; then
+    size=$(echo $sudoPW | sudo -S blockdev --getsize64 ${target_dev})
+  else
+    size=$2
+  fi
+
+  heads=255
+  sectors=63
+  sector=512
+  cylinders=$(( ${size} / ${heads} / ${sectors} / ${sector} ))
+  rootfs_size=$(( $1 - 1 ))
+  pboot_offset=$(( ${pboot_size} - 1 ))
+  primary_start=$(( ${pboot_start} + ${pboot_offset} + 1 ))
+  secondary_start=$(( ${primary_start} + ${rootfs_size} + 1 ))
+  data_start=$(( ${secondary_start} + ${rootfs_size} + 1 ))
+  data_offset=$(( ${pdata_size} - 1 ))
+
+  echo "3. Format device... size for rootfs: $1"
+
+  sed -e 's/\s*\([\+0-9a-zA-Z]*\).*/\1/' << EOF | sudo fdisk ${target_dev}
+	o # clear the in memory partition table
+	x
+	h
+	${heads}
+	s
+	${sectors}
+	c
+	${cylinders}
+	r
+	n # new partition
+	p # primary partition
+	1 # partition number 1
+	${pboot_start}  # default - start at beginning of disk 
+	+${pboot_offset} # 16 MB boot parttion
+	t
+	c
+	a
+	n # new partition
+	p # primary partition
+	2 # partion number 2
+	${primary_start}	# start immediately after preceding partition
+	+${rootfs_size}
+	n # new partition
+	p # primary partition
+	3 # partion number 3
+	${secondary_start}	# start immediately after preceding partition
+	+${rootfs_size}
+	n # new partition
+	p # primary partition
+	${data_start}		# start immediately after preceding partition
+	+${data_offset}
+	p # print the in-memory partition table
+	w # write the partition table
+	q # and we're done
+EOF
+}
+
+make_filesystem() {
+  partitions=($(echo $sudoPW | sudo -S fdisk -l ${target_dev} | grep '^/dev' | cut -d' ' -f1))
+
+  number_of_parts=${#partitions[@]}
+
+  [[ $number_of_parts -eq 4 ]] || { echo "Error: incorect device partitioning."; exit 1; } 
+
+  echo "4. Make filesystem for $number_of_parts partitions..."
+
+  for partition in "${partitions[@]}"
+  do
+    echo "Formatting partition: ${partition}..."
+    part_no=$(get_part_number_from_device $partition)
+
+    if [[ part_no -eq 1 ]]; then
+      echo $sudoPW | sudo -S mkfs.vfat -n ${part_names[${part_no} - 1]} $partition
+    else
+      echo $sudoPW | sudo -S mkfs.ext4 -L ${part_names[${part_no} - 1]} -F $partition
+    fi
+  done
+}
+
+make_filesystem_sdimg() {
```

Comment:
What is the difference between this and `make_filesystem`? Seems like a lot of duplication

Reply:
Will be fixed.